pkgs/by-name/do/docmost/bcrypt-lib.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  node-pre-gyp,
+}:
+let
+  version = "5.1.1";
+in
+buildNpmPackage {
+  pname = "bcrypt-lib";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "kelektiv";
+    repo = "node.bcrypt.js";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-mgfYEgvgC5JwgUhU8Kn/f1D7n9ljnIODkKotEcxQnDQ=";
+  };
+
+  npmDepsHash = "sha256-CPXZ/yLEjTBIyTPVrgCvb+UGZJ6yRZUJOvBSZpLSABY=";
+
+  nativeBuildInputs = [
+    node-pre-gyp
+  ];
+
+  npmBuildScript = "install";
+
+  buildPhase = ''
+    runHook preBuild
+    node-pre-gyp install --build-from-source
```

Comment:
Upstream relies on `node-gyp-build` which in turn relies on `prebuildify`. Using that would allow us to use `BCRYPT_PREBUILD` to define the path to our own builds. We could even use a setup hook for that.

Reply:
Can you elaborate 😅

Reply:
So upstream uses https://github.com/prebuild/node-gyp-build to manage its native library, which supports overriding the path to the prebuilt library using `${name}_PREBUILD` (see https://github.com/prebuild/node-gyp-build/blob/6822ec52423a2b3ed48ef8960a9fe05902e9e1a3/node-gyp-build.js#L30). You can therefore utilize the tool `prebuildify` from the same vendor to build the native library and then define `env.BCRYPT_PREBUILD` in the main derivation to point to it. For reference, upstream uses prebuildify too: https://github.com/kelektiv/node.bcrypt.js/blob/d8195c6c4bb170b3b2ee2d04047a60fda80aa091/package.json#L30